### PR TITLE
Integrating font awesome and other Icon libraries

### PR DESCRIPTION
**Pull request by dbd5**
Hi @unclead ,

Please review this PR for integrating font awesome or other icon libraries into yii2-multiple-input

Regards

### CODE REVIEW

File: src/renderers/TableRenderer.php [hunk 1]
```diff
@@ -253,9 +253,17 @@ public function renderCellContent($column, $index)
...
-            'id' => $id
-        ]);
+
+        /**
+         * This class inherits fontMap from BaseRenderer
+         * If the input to be rendered is a drag column, we give it the appropriate icon class
+         * via the $options array
+         */
+        $options = ['id' => $id];
+        if (substr($id, -4)=='drag') {
```
Comment (unclead) [inline src/renderers/TableRenderer.php]: need space before and after `==` and why not `===`?

File: src/renderers/TableRenderer.php [hunk 2]
```diff
@@ -253,9 +253,17 @@ public function renderCellContent($column, $index)
...
+            $options = ArrayHelper::merge($options, ['class'=>$this->fontMap['drag-handle']]);
```
Comment (unclead) [inline src/renderers/TableRenderer.php]: add space before and after `=>`

File: src/renderers/BaseRenderer.php [hunk 3]
```diff
@@ -489,4 +491,15 @@ protected function prepareJsAttributes()
 
         return $attributes;
     }
+
+    /**
+     * @param $action - the control parameter, used as key into allowed types
+     * @return string - the relevant icon class
+     */
+    protected function getFontClass($action) {
```
Comment (unclead) [inline src/renderers/BaseRenderer.php]: may be throw an exception if tried to get class for inappropriate action?

File: src/components/BaseColumn.php [hunk 4]
```diff
@@ -467,7 +467,18 @@ protected function renderStatic($name, $value, $options)
      */
     protected function renderDragColumn($name, $value, $options)
     {
-        return Html::tag('span', null, ['class' => 'glyphicon glyphicon-menu-hamburger drag-handle']);
+        // return Html::tag('span', null, ['class' => 'glyphicon glyphicon-menu-hamburger drag-handle']);
```
Comment (unclead) [inline src/components/BaseColumn.php]: forgot to remove?

File: src/MultipleInput.php [hunk 5]
```diff
@@ -255,7 +279,11 @@ private function createRenderer()
 
             array_unshift($this->columns, $drag);
         }
-        
+
+        /**
+         * set default font map
+         */
+        $fontmap = array_key_exists($this->fontSource, $this->fontMap)?$this->fontMap[$this->fontSource]:$this->fontMap['glyphicons'];
```
Comment (unclead) [inline src/MultipleInput.php]: add spaces between operands for more readability

File: src/TabularInput.php [hunk 6]
```diff
@@ -200,6 +224,10 @@ public function run()
      */
     private function createRenderer()
     {
+        /**
+         * set default font map
+         */
+        $fontmap = array_key_exists($this->fontSource, $this->fontMap)?$this->fontMap[$this->fontSource]:$this->fontMap['glyphicons'];
```
Comment (unclead) [inline src/TabularInput.php]: spaces

**Comment by unclead** [comment]
@dbd5 looks good, some minor notes about the code format. Fix them and I will merge the PR

**Comment by dbd5** [comment]
Hi @unclead ,

I've made the recommended changes. I've also renamed some variables like fontMap to iconMap and fontSource to iconSource to better reflect the objective of the PR

**Pull request merged**